### ISSUE TITLE
Getting: nb: fatal: TypeError:

### ISSUE THREAD

**Issue by SirColeslaw** [ebaauw/homebridge-nb#4]
Hello,

im getting this Error all the time for a fresh homebrindge-nb installation.

homebridge@ubuntu:/var/lib/homebridge$ nb discover
nb: fatal: TypeError: Cannot read property 'split' of undefined
    at Main.parseArguments (/usr/lib/node_modules/homebridge-nb/cli/nb.js:328:29)
    at Main.main (/usr/lib/node_modules/homebridge-nb/cli/nb.js:336:27)
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-nb/cli/nb.js:586:12)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47

It seems that no matter which command I use, the same error always occurs.

**Comment by ebaauw** [comment]
That’s not good.  As a workaround, could you try `nb -H x discover?  For other commands, you need to specify the ip address of your nuki bridge, but `discover` shouldn’t look at the value.

**Comment by SirColeslaw** [comment]
I tried that too, but unfortunately the message remains the same.

**Comment by ebaauw** [comment]
> I tried that too, but unfortunately the message remains the same.

Huh?  What version did you install?
```
$ nb discover
nb: fatal: TypeError: Cannot read property 'split' of undefined
    at Main.parseArguments (/Users/ebaauw/GitHub/homebridge-nb/cli/nb.js:328:29)
    at Main.main (/Users/ebaauw/GitHub/homebridge-nb/cli/nb.js:336:27)
    at Object.<anonymous> (/Users/ebaauw/GitHub/homebridge-nb/cli/nb.js:586:12)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
$ nb -H x discover
[
  {
    "bridgeId": 1522130723,
    "dateUpdated": "2020-11-23T18:26:50Z",
    "ip": "192.168.76.201",
    "port": 8080
  }
]
```

Can you please try:
```
$ nb -DD -H x discover
nb discover: nuki server request 1: GET /bridges
nb discover: nuki server request 1: GET https://api.nuki.io/discover/bridges
nb discover: nuki server request 1: response: {"bridges":[{"bridgeId":1522130723,"ip":"192.168.76.201","port":8080,"dateUpdated":"2020-11-23T18:26:50Z"}],"errorCode":0}
nb discover: nuki server request 1: 200 OK
[
  {
    "bridgeId": 1522130723,
    "dateUpdated": "2020-11-23T18:26:50Z",
    "ip": "192.168.76.201",
    "port": 8080
  }
]
```

> If it helps, every 10 minutes I get this entry in the status log of Homebridge:

I don't think that's related to the `nb` bug.  I would theorise Homebridge NB chokes on the response from the Nuki portal, trying to discover your Nuki bridge(s).  I need to see the output of `ph discover` to understand what's happening here.  The stack trace suggests you're not running the latest version of Homebridge NB?

**Comment by SirColeslaw** [comment]
Version installed is: homebridge-nb v0.0.5

I tried a downgrade before contacting you to look if it solves the problem, but it didnt so i updated back to 0.0.5.

Here is the output from nb -DD -H x Discover:

homebridge@ubuntu:/var/lib/homebridge$ nb -DD -H x discover
nb discover: nuki server request 1: GET /bridges
nb discover: nuki server request 1: GET https://api.nuki.io/discover/bridges
nb discover: nuki server request 1: response: {"bridges":[{"bridgeId":176530759,"dateUpdated":"2020-11-30T14:29:29Z"}],"errorCode":0}
nb discover: nuki server request 1: 200 OK
[
  {
    "bridgeId": 176530759,
    "dateUpdated": "2020-11-30T14:29:29Z"
  }
]

**Comment by ebaauw** [comment]
That is weird, the response from the Nuki portal doesn't include your bridge's local IP address.  I didn't know it could report a bridge without one (I would have expected an empty list instead).

You need to enable something on the Nuki bridge for it to be discoverable.

**Comment by ebaauw** [comment]
v0.0.6 should fix the TypeErrors.

**Comment by SirColeslaw** [comment]
Could it be a problem that i am using the nuki software bridge via android instead of a hardware one? On that software bridge i dont have any button i could push to make it discoverable.

**Comment by ebaauw** [comment]
I have no idea.  I read about the software bridge in the API documentation, but I haven’t been able to find it for download.  Does it run only on Android?  The button on the hardware bridge is to obtain the API token, not to make it discoverable.

**Comment by SirColeslaw** [comment]
Software bridge seems to be Android only: https://play.google.com/store/apps/details?id=io.nuki.bridge&hl=de&gl=US

There is an option to generate a token on the nuki web interface. Can that be used to insert that generated token into the plugin somehow?

![Unbenannt](https://user-images.githubusercontent.com/45994510/100838231-60360080-3472-11eb-93ab-8b851e1f889a.PNG)

**Comment by ebaauw** [comment]
You can feed the token (and the bridge’s IP address) to `nb` by setting `NB_TOKEN` (and `NB_HOST`) or through command line parameters.  There’s currently no way to feed these to Homebridge NB.

Could you check that `nb` can interact with the software bridge, controlling your lock?  Does the software bridge support callbacks?

**Comment by SirColeslaw** [comment]
I found out that the Token can be received on the software bridge app under settings, so i have that token now, but i dont know now where to put it.

**Comment by ebaauw** [comment]
You cannot put a token nor hostname to Homebridge NB.  It's all discovered automatically.  Below is what happens on the hardware bridge.  For Homebridge NB to support the software bridge, it would have to behave similar to the hardware bridge.

You need to enable the HTTP API and API Quick discovery on your bridge, in the Nuki app:
![IMG_0232](https://user-images.githubusercontent.com/22179355/101193853-7227c700-365d-11eb-9d14-d18e99527a25.png)

After that, `nb discover` finds the bridge:
```
$ nb discover
[
  {
    "bridgeId": 1522130723,
    "dateUpdated": "2020-12-04T17:19:29Z",
    "ip": "xxx.xxx.xxx.xxx",
    "port": 8080
  }
]
```

Homebridge NB does an `/auth` API call to the bridge to obtain the token.  As security measure, you need to press the button on the bridge.

The discovered `bridgeId` is actually the `serverId` returned by an API call to `/info`:
```
$ nb info
{
  "bridgeType": 1,
  "currentTime": "2020-12-04T17:23:52+00:00",
  "ids": {
    "hardwareId": 535158466,
    "serverId": 1522130723
  },
  "scanResults": [
    {
      "deviceType": 2,
      "name": "Nuki_Opener_1E60104A",
      "nukiId": 509612106,
      "paired": true,
      "rssi": -40
    },
    {
      "deviceType": 0,
      "name": "Nuki_1FDE0EED",
      "nukiId": 534646509,
      "paired": true,
      "rssi": -76
    }
  ],
  "serverConnected": true,
  "uptime": 274,
  "versions": {
    "firmwareVersion": "2.8.0",
    "wifiFirmwareVersion": "2.2.0"
  },
  "wlanConnected": true
}
```
According to the HTTP API documentation, the only differences with the software bridge would be a missing `hardwareId` and a `bridgeType` of 2.  I have no idea how you'd simulate pressing the button on the software bridge.

**Comment by ebaauw** [comment]
The `TypeError`s should be fixed in v0.0.7.
